integrated_channels/integrated_channel/learner_data.py

```diff
@@ -134,6 +133,34 @@ def collect_learner_data(self):
                 is_passing=is_passing,
             )
 
+    def _should_skip_enrollment(self, enterprise_enrollment):
+        """
+        Specifies whether to skip data collection of the enterprise enrollment due to any configuration reason.
+
+        :param enterprise_enrollment: The enterprise enrollment whose configuration is to be checked against.
+        :return: True if we should skip, False otherwise.
+        """
+        return any([
+            not enterprise_enrollment.consent_available,
+            self._disabled_audit_track_data_reporting(enterprise_enrollment),
+        ])
+
+    def _disabled_audit_track_data_reporting(self, enterprise_enrollment):
+        """
+        Specifies whether to skip data collection due to audit track data reporting not being enabled.
+
+        :param enterprise_enrollment: The enterprise enrollment whose reporting configuration is to be checked against.
+        :return: True if we should skip, False otherwise.
+        """
+        if not self.enterprise_customer.enables_audit_data_reporting:
+            if self.course_enrollment_api is None:
+                self.course_enrollment_api = EnrollmentApiClient()
+            course_enrollment = self.course_enrollment_api.get_course_enrollment(
+                enterprise_enrollment.enterprise_customer_user.username,
+                enterprise_enrollment.course_id
+            )
+            return course_enrollment['mode'] in ['audit', 'honor'] if course_enrollment is not None else True
```

Comment:
Thanks for noting these points; I didn't like this function the way it currently is either but was hoping your review could help out. I'll refactor this to make it more readable!